$OpenBSD: patch-core_byteswap_h,v 1.2 2020/11/09 21:48:25 cwen Exp $

Hunk #1: Fix the build on powerpc, upstream code has been totally changed
Hunk #2: Extracted from https://github.com/JulNadeauCA/libagar/commit/56b3500

Index: core/byteswap.h
--- core/byteswap.h.orig
+++ core/byteswap.h
@@ -33,15 +33,17 @@ AG_Swap16(Uint16 x)
 		"0" (x));
 	return (x);
 }
-#elif defined(__GNUC__) && (defined(__powerpc__) || defined(__ppc__))
+#elif (defined(__GNUC__) || defined(__clang__)) && \
+      (defined(__powerpc__) || defined(__ppc__)) && \
+     (!defined(__powerpc64__) && !defined(__ppc64__))
 static __inline__ Uint16
 AG_Swap16(Uint16 x)
 {
-	Uint16 rv;
+	Uint32 rv;
 	__asm__("rlwimi %0,%2,8,16,23" :
 	        "=&r" (rv) :
 		"0" (x >> 8), "r" (x));
-	return (rv);
+	return (Uint16) (rv);
 }
 #else
 static __inline__ Uint16
@@ -72,7 +74,9 @@ AG_Swap32(Uint32 x)
 		"0" (x));
 	return (x);
 }
-#elif defined(__GNUC__) && (defined(__powerpc__) || defined(__ppc__))
+#elif (defined(__GNUC__) || defined(__clang__)) && \
+      (defined(__powerpc__) || defined(__ppc__)) && \
+     (!defined(__powerpc64__) && !defined(__ppc64__))
 static __inline__ Uint32
 AG_Swap32(Uint32 x)
 {
